tests/unit/test_is_stuck.py

```diff
@@ -0,0 +1,224 @@
+from unittest.mock import Mock, patch
```

Comment:
I was about to say "it'd be great to get some unit tests in here" and then I scrolled down a bit 😄 

Thanks for this!